pyomo/contrib/ampl_function_demo/tests/test_ampl_function_demo.py

```diff
@@ -0,0 +1,34 @@
+import platform
+import pyomo.environ as pyo
+import pyomo.common.unittest as unittest
+from pyomo.core.base.external import ExternalFunction
+from pyomo.common.fileutils import find_library
+from pyomo.opt import check_available_solvers
+
+flib = find_library("function_ASL")
+
+class TestAMPLExternalFunction(unittest.TestCase):
+    def test_eval_function(self):
+        if not flib:
+            self.skipTest("Could not find the function_ASL.dll library")
+        if platform.python_implementation().lower().startswith('pypy'):
+            self.skipTest("AMPLExternalFunction interface does not work on pypy")
```

Comment:
Why don't you need this same `skipTest` check for the other test in this file (`test_solve_function`)?

Reply:
I think you can't evaluate the function from pypy, but you can send it to the solver without issue.